Draft-Accepted/feedback_completer_auto_discovery.md

```diff
@@ -0,0 +1,415 @@
+   Will that be a problem? How to run it in the top-level session instead?
+   - [dongbo] This is more of a implementation-detail question. It's not a problem for module thanks to https://github.com/PowerShell/PowerShell/pull/24909.
+
+3. Today, a user can unregister a tab-completer by running `Register-ArgumentCompleter -CommandName <name> -ScriptBlock $null`.
+   But it doesn't remove the `<name>` from cache, but only sets its value to `null`. We probably should fix it to make our cache lookup simpler.
+
+4. Today, a user cannot see what commands have tab-completer registered. This may be out-of-scope for this RFC.
+
+### When to enable/disable the feature
+
+This feature is only for interactive session, so we need to decide on when the feature is enabled and when is disabled.
+
+1. The feature only works when the PowerShell host is `ConsoleHost` or the `Visual Studio Code Host` (which uses `ConsoleHost` internally IIRC).
+   - Today, only `ConsoleHost` consumes feedback providers.
+2. The feature should be disabled when PowerShell starts with `-Command` or `-File` to run a command or a file and then terminates.
+   - So, when `-noexit` is specifeid, the feature should be enabled if other conditions are satisfied.
+2. Shall we disable the feature with the `-noninteractive` flag?
+   - `PSReadLine` is disabled when this flag is specified, so maybe this feature should be disabled too.
+3. Shall we disable the feature with the `-noprofile` flag?
+   - The "loading at startup" part of the feature is similar to how profiles are processed, but it's not part of the profile.
+4. Shall we add a flag (or flags) to allow user to disable this feature (or disable `feedback` and `completer` separately)?
+5. We report progress when loading `feedback` or `completer` at startup, so how to allow users to disable the progress report?
+   - We have the `-NoProfileLoadTime` flag today to not show the time taken for running profile.
+6. How about on a System Lockdown Mode (SLM) or Restricted remoting environments?
+   - We use `.psd1` file for metadata, which can be signed if needed.
+
+### Unified Location for load-at-startup Configurations
+
+There could be a similar demand for a predictor module. There won't be a specific trigger for any predictors,
+so for a predictor to be auto-discovered, it has to be loaded at the startup of an interactive session.
+
+Given that, maybe it's better to have a unified location for all the load-at-startup configurations:
+
+- Have a `startup` folder at the same level of `feedbacks` and `completions` folders;
+- All modules or scripts that need to be processed at session startup should have configurations deployed in the `startup` folder.
+
+Each item within `startup` is a folder, whose name should be the friendly name of the component, e.g. `"UnixTabCompletion"`.
+Within each sub-folder, a `.psd1` file named after the folder name should be defined to configure the auto-discovery of the component.
```

Comment:
As I mentioned in a comment above, this sounds exactly like the `Modules` folder, except with a different manifest structure. I'd prefer to reuse modules for this rather than defining another mechanism.